test/AllocatorWithLimitTest.cpp

```diff
@@ -10,28 +10,29 @@
 
 using ad_utility::AllocatorWithLimit;
 using ad_utility::makeAllocationMemoryLeftThreadsafeObject;
+using namespace ad_utility::memory_literals;
 
 using V = std::vector<int, AllocatorWithLimit<int>>;
 TEST(AllocatorWithLimit, initial) {
   AllocatorWithLimit<int> all{
-      ad_utility::makeAllocationMemoryLeftThreadsafeObject(2ul << 20)};
+      ad_utility::makeAllocationMemoryLeftThreadsafeObject(2_MB)};
   static_assert(sizeof(int) == 4);
   [[maybe_unused]] auto ptr = all.allocate(250'000);
-  ASSERT_EQ(all.numFreeBytes(), (2ul << 20) - 1'000'000);
-  ASSERT_EQ(std::as_const(all).numFreeBytes(), (2ul << 20) - 1'000'000);
+  ASSERT_EQ(all.amountMemoryLeft(), 1_MB);
+  ASSERT_EQ(std::as_const(all).amountMemoryLeft(), 1_MB);
   try {
-    all.allocate(600'000);
+    all.allocate(500'000);
     FAIL() << "Should have thrown";
   } catch (const ad_utility::detail::AllocationExceedsLimitException& e) {
     ASSERT_THAT(e.what(),
                 ::testing::StartsWith(
-                    "Tried to allocate 2MB, but only 1MB were available."));
+                    "Tried to allocate 2 MB, but only 1 MB were available."));
   }
```

Comment:
We in the meantime have `AD_EXPECT_THROW_WITH_MESSAGE` in the GTestHelpers file, this can be used to simplify such tests.

Reply:
I'm aware. However, improving already existing tests should be a separate branch.